### ISSUE TITLE
kmc: Cannot tap homebrew/science: invalid syntax in tap!

### ISSUE THREAD

**Issue by lordqwerty** [Homebrew/homebrew-science#5379]
### Indicate troubleshooting steps taken

- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [X] Checked that a similar issue hasn't already been filed?
- [ ] Looked up the software bug tracker if reporting a build error?
- [ ] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

#### Brief summary of issue:

Unable to install `maude` because of `"cxx14" is not a recognized standard` in `kmc`

#### The complete command that you issued:

```
$ brew install homebrew/science/maude
==> Tapping homebrew/science
Cloning into '/home/jorden/.linuxbrew/Library/Taps/homebrew/homebrew-science'...
remote: Counting objects: 687, done.
remote: Compressing objects: 100% (682/682), done.
remote: Total 687 (delta 0), reused 52 (delta 0), pack-reused 0
Receiving objects: 100% (687/687), 626.77 KiB | 0 bytes/s, done.
Checking connectivity... done.
Error: Invalid formula: /home/jorden/.linuxbrew/Library/Taps/homebrew/homebrew-science/kmc.rb
"cxx14" is not a recognized standard
Error: Cannot tap homebrew/science: invalid syntax in tap!
```

#### Gist with output of `brew doctor`:

```
Your system is ready to brew.
```

#### Output of `brew config`:

```
HOMEBREW_VERSION: 1.1.10
ORIGIN: https://github.com/Linuxbrew/brew
HEAD: d8c8b867bf99c604c89bc848f58d1fae8afd599c
Last commit: 6 weeks ago
Core tap ORIGIN: https://github.com/Linuxbrew/homebrew-core
Core tap HEAD: 529df9804b1b09bfda0be829b5c8def42af44eeb
Core tap last commit: 3 hours ago
HOMEBREW_PREFIX: /home/jorden/.linuxbrew
HOMEBREW_REPOSITORY: /home/jorden/.linuxbrew
HOMEBREW_CELLAR: /home/jorden/.linuxbrew/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://linuxbrew.bintray.com
CPU: quad-core 64-bit sandybridge
Homebrew Ruby: 2.3.3 => /usr/bin/ruby
Clang: N/A
Git: 2.9.3 => /usr/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python => /usr/bin/python2.7
Ruby: /usr/bin/ruby
Java: 1.8.0_121
Kernel: Linux 4.10.6-200.fc25.x86_64 x86_64 GNU/Linux
OS: Fedora release 25 (Twenty Five)
Codename: TwentyFive
OS glibc: 2.24
OS gcc: 6.3.1
Linuxbrew glibc: N/A
Linuxbrew gcc: N/A
Linuxbrew xorg: N/A
```

**Comment by lordqwerty** [comment]
@ilovezfs it appears this was done in https://github.com/Homebrew/homebrew-science/commit/2d00e99d095d120fb43e34d839e086e5daf35554 . However this does not work with Linuxbrew. Cannot tap formula. 

@sjackman are you experiencing this same issue?

**Comment by lordqwerty** [comment]
Closing it was a `$PATH` variable issue.

**Comment by heuermh** [comment]
I'm seeing the same with linuxbrew in docker at DockerHub, e.g.
https://hub.docker.com/r/heuermh/adam/builds/bbfi234i2t6fbpueostorzg/
https://hub.docker.com/r/heuermh/adam/builds/bdxivkuwkkoy8xq48upzxtl/

What is the `$PATH` variable issue?

**Comment by lordqwerty** [comment]
Can you past the output of `brew config` and `brew doctor`? 
For me in the `.bash_profile` file adding: 
```
fpath=($PATH)
```

Prior to:
```
export PATH="$HOME/.linuxbrew/bin:$PATH"
export MANPATH="$HOME/.linuxbrew/share/man:$MANPATH"
export INFOPATH="$HOME/.linuxbrew/share/info:$INFOPATH"
```

Once complete run `exec $SHELL && source $HOME/.bashrc` should resolve your issue.

**Comment by sjackman** [comment]
Linuxbrew 1.1.10 doesn't support the line `needs :cxx14` in the formula `homebrew/science/kmc`. You can update `brew` to master, or you can wait for us to release Linuxbrew 1.1.11, which should be happening quite soon.

**Comment by heuermh** [comment]
Thank you, @sjackman @lordqwerty for the explanation.

I'm at the mercy of `sjackman/linuxbrew`, so I'll pick up the fix when it makes its way there.

**Comment by sjackman** [comment]
The Docker image `linuxbrew/linuxbrew` is the one to use. It already has this fix.

**Comment by sadransh** [comment]
@sjackman I am a little bit starter could you please tell me how to do that? 
I didn't find exactly "The Docker image linuxbrew/linuxbrew"  and how to install it on my current linuxbrew

**Comment by alanhoyle** [comment]
@sjackman, there are a number of Dockerfiles over at https://github.com/heuermh/docker-linuxbrew-bio that seem to be dependent on your sjackman/linuxbrew.  I tried forking that repo and substituting "linuxbrew/linuxbrew" for "sjackman/linuxbrew" but I still get the same error there on the ones I tried (freebayes and snpeff, e.g).

>==> Tapping homebrew/science
Cloning into '/home/linuxbrew/.linuxbrew/Library/Taps/homebrew/homebrew-science'...
Error: Invalid formula: /home/linuxbrew/.linuxbrew/Library/Taps/homebrew/homebrew-science/kmc.rb
Error: Cannot tap homebrew/science: invalid syntax in tap!
"cxx14" is not a recognized standard
